src/logic.ts

```diff
@@ -94,3 +97,38 @@ export function sortBreadcrumbs(
   });
   return orderedBreadcrumbs;
 }
+
+export function findNextNodeOfType({
+  flow,
+  breadcrumbs,
+  componentType,
+}: {
+  flow: FlowGraph;
+  breadcrumbs: Breadcrumbs;
+  componentType: ComponentType;
+}): IndexedNode | undefined {
+  const orderedBreadcrumbs = sortBreadcrumbs(flow, breadcrumbs);
+  const lastCrumb = orderedBreadcrumbs.at(-1)!;
+  const sortedFlow = sortFlow(flow);
+  const sortedFlowLastNodeIndex = sortedFlow.findIndex(
+    (n) => n.id == lastCrumb.id
+  );
+  const truncatedFlow = sortedFlow.slice(sortedFlowLastNodeIndex);
+
+  for (const node of truncatedFlow) {
+    const parentNodeIsLastCrumb = node.parentId === lastCrumb.id;
+    if (parentNodeIsLastCrumb && node.type === componentType) {
+      return node;
+    }
+    const parentNodeIsCrumbAnswer =
+      node.parentId &&
+      lastCrumb.answers &&
+      lastCrumb.answers.includes(node.parentId!);
+    if (parentNodeIsCrumbAnswer && node.type === componentType) {
+      return node;
+    }
+  }
+  // when the above fails, fallback to a naïve scan of matching upcoming nodes
+  // FIXME: this approach does not respect flow branching logic
+  return truncatedFlow.find((n) => n.type === componentType);
```

Comment:
This function is sufficient for invite-to-pay use cases (looking for the Send component which is always just after the last breadcrumb) but falls-back to an imperfect hack where the next node doesn't match. This may not work in every case so I'd recommend we use this function carefully until then.

Reply:
I do have a plan for introducing more thorough node traversal logic in the near future (time permitting) so I will come back to this with a more robust approach.